src/main/kotlin/org/jitsi/nlj/rtp/bandwidthestimation/GoogleCcEstimator.kt

```diff
@@ -0,0 +1,113 @@
+    }
+
+    private val logger = parentLogger.createChildLogger(GoogleCcEstimator::class)
+
+    /**
+     * Implements the delay-based part of Google CC.
+     */
+    private val bitrateEstimatorAbsSendTime = RemoteBitrateEstimatorAbsSendTime(diagnosticContext, logger)
+    init {
+        bitrateEstimatorAbsSendTime.setMinBitrate(minBw.bps.toInt())
+    }
+
+    /**
+     * Implements the loss-based part of Google CC.
+     */
+    private val sendSideBandwidthEstimation = SendSideBandwidthEstimation(diagnosticContext, initBw.bps.toLong(), logger)
+    init {
+        sendSideBandwidthEstimation.setMinMaxBitrate(minBw.bps.toInt(), maxBw.bps.toInt())
+    }
+
+    override fun processPacketArrival(now: Instant, sendTime: Instant?, recvTime: Instant?, seq: Int, size: DataSize, ecn: Byte) {
+        super.processPacketArrival(now, sendTime, recvTime, seq, size, ecn)
+
+        if (sendTime != null && recvTime != null) {
+            bitrateEstimatorAbsSendTime.incomingPacketInfo(now.toEpochMilli(),
+                    recvTime.toEpochMilli(), sendTime.toEpochMilli(), size.bytes.toInt(), 1 /* TODO */)
+        }
+        sendSideBandwidthEstimation.updateReceiverEstimate(bitrateEstimatorAbsSendTime.latestEstimate)
+        sendSideBandwidthEstimation.updateReceiverBlock(0, 1, now.toEpochMilli())
+
+        /* TODO: rate-limit how often we call updateEstimate? */
+        sendSideBandwidthEstimation.updateEstimate(now.toEpochMilli())
+        reportBandwidthEstimate(now, sendSideBandwidthEstimation.latestEstimate.bps)
+    }
+
+    override fun processPacketLoss(now: Instant, sendTime: Instant?, seq: Int) {
+        super.processPacketLoss(now, sendTime, seq)
+
+        sendSideBandwidthEstimation.updateReceiverBlock(256, 1, now.toEpochMilli())
+        sendSideBandwidthEstimation.updateEstimate(now.toEpochMilli())
+        reportBandwidthEstimate(now, sendSideBandwidthEstimation.latestEstimate.bps)
+    }
+
+    override fun onRttUpdate(now: Instant, newRtt: Duration) {
+        super.onRttUpdate(now, newRtt)
+
+        bitrateEstimatorAbsSendTime.onRttUpdate(now.toEpochMilli(), newRtt.toMillis(), -1)
+        sendSideBandwidthEstimation.onRttUpdate(newRtt.toNanos() / 1.0e9)
+    }
+
+    override fun getCurrentBw(now: Instant): Bandwidth {
+        return sendSideBandwidthEstimation.latestEstimate.bps
+    }
+
+    override fun getStats(): NodeStatsBlock = NodeStatsBlock("GoogleCcEstimator").apply {
+        addNumber("latestLossEstimate", sendSideBandwidthEstimation.latestREMB)
+        addNumber("latestEstimate", sendSideBandwidthEstimation.latestEstimate)
+        addNumber("latestFractionLoss", sendSideBandwidthEstimation.latestFractionLoss)
+        val bweStats = sendSideBandwidthEstimation.statistics
+        addNumber("lossDegradedMs", bweStats.lossDegradedMs)
+        addNumber("lossFreeMs", bweStats.lossFreeMs)
+        addNumber("lossLimitedMs", bweStats.lossLimitedMs)
```

Comment:
this could be:
```
with (sendSideBandwidthEstimation.statistics) {
        addNumber("lossDegradedMs", lossDegradedMs)
        addNumber("lossFreeMs", lossFreeMs)
        addNumber("lossLimitedMs", lossLimitedMs)
}
```